### PR TITLE
Fix Series v Index bool ops

### PR DESCRIPTION
**Pull request by jbrockmendel**
Series boolean ops tests need some serious cleanup and fleshing out.  This gets separates out one fix before going in to clean up the tests.

### CODE REVIEW

File: pandas/core/ops.py [hunk 1]
```diff
@@ -1448,23 +1448,22 @@ def _bool_method_SERIES(cls, op, special):
...
-                y = construct_1d_object_array_from_listlike(y)
-
-            if isinstance(y, (np.ndarray, ABCSeries)):
-                if (is_bool_dtype(x.dtype) and is_bool_dtype(y.dtype)):
-                    result = op(x, y)  # when would this be hit?
-                else:
-                    x = ensure_object(x)
-                    y = ensure_object(y)
-                    result = libops.vec_binop(x, y, op)
+            assert not isinstance(y, (list, ABCSeries, ABCIndexClass))
```
Comment (gfyoung) [inline pandas/core/ops.py]: From an error reporting perspective, I might go for:

~~~
if isinstance(y, (list, ABCSeries, ABCIndexClass)):
   raise
~~~

Comment (gfyoung) [reply pandas/core/ops.py]: This comment BTW applies to any of your other `assert` statements.

File: pandas/tests/series/test_operators.py [hunk 2]
```diff
@@ -1433,6 +1433,20 @@ def test_ops_datetimelike_align(self):
         result = (dt2.to_frame() - dt.to_frame())[0]
         assert_series_equal(result, expected)
 
+    def test_bool_ops_with_index(self):
+        # GH#19792
+        # TODO: reversed ops still raises, GH#22092
```
Comment (gfyoung) [inline pandas/tests/series/test_operators.py]: Can we have an `xfail` test for that in this PR?

(unless you plan to patch the reversed op in the near term, then not needed)

Comment (jbrockmendel) [reply pandas/tests/series/test_operators.py]: +1 on this idea.  I'll wait til AM to implement it, ideally on the follow-up to #22153.

**Comment by jreback** [comment]
let me have another look, can you rebase

**Comment by jbrockmendel** [comment]
Rebased+green

File: pandas/core/ops.py [hunk 3]
```diff
@@ -1561,33 +1560,40 @@ def wrapper(self, other):
...
+            if is_list_like(other) and not isinstance(other, np.ndarray):
+                # TODO: Can we do this before the is_integer_dtype check?
+                # could the is_integer_dtype check be checking the wrong
+                # thing?  e.g. other = [[0, 1], [2, 3], [4, 5]]?
+                other = construct_1d_object_array_from_listlike(other)
+
+            ovalues = other
+            finalizer = lambda x: x.__finalize__(self)
+
+        filler = (fill_int if is_self_int_dtype and is_other_int_dtype
```
Comment (jreback) [inline pandas/core/ops.py]: can you add a comment on what is going on with the filler

File: pandas/tests/series/test_operators.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/series/test_operators.py]: can you add this test?

**Review by jreback — CHANGES_REQUESTED** [review]
small comments, pls rebase

**Comment by jreback** [comment]
doesn't this overlap with #22293 ?

**Comment by jbrockmendel** [comment]
Yep, commented to that effect over there.

**Comment by jreback** [comment]
ok then, merge that one first?

**Comment by jbrockmendel** [comment]
>ok then, merge that one first?

Sure.  After rebase this'll be a small follow-up

File: pandas/tests/series/test_operators.py [hunk 4]
```diff
@@ -627,6 +604,42 @@ def test_ops_datetimelike_align(self):
         result = (dt2.to_frame() - dt.to_frame())[0]
         assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize('op', [
+        operator.and_,
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: pls add these as a fixture in conftest (follow-on PR)

**Review by jreback — APPROVED** [review]

**Pull request merged**